xarray/core/groupby.py

```diff
@@ -338,7 +338,7 @@ def _factorize_bins(
     if (codes == -1).all():
         raise ValueError(f"None of the data falls within bins with edges {bins!r}")
     full_index = binned.categories
-    unique_values = binned.unique().dropna()
+    unique_values = np.sort(binned.unique().dropna())
```

Comment:
```suggestion
    unique_values = np.unique(binned.dropna())
```
Just an alternative, I haven't compared performances.

unique_values has changed type from categorical to np.ndarray but maybe it's ok?

```python
a, b = pd.cut([4,3,2,1,3,0, 4], [0,2,4], retbins=True)
a
Out[96]: 
[(2.0, 4.0], (2.0, 4.0], (0.0, 2.0], (0.0, 2.0], (2.0, 4.0], NaN, (2.0, 4.0]]
Categories (2, interval[int64, right]): [(0, 2] < (2, 4]]

type(a)
Out[104]: pandas.core.arrays.categorical.Categorical

unique_values = np.sort(a.unique().dropna())
unique_values
Out[106]: 
array([Interval(0, 2, closed='right'), Interval(2, 4, closed='right')],
      dtype=object)

np.unique(a.dropna())
Out[108]: 
array([Interval(0, 2, closed='right'), Interval(2, 4, closed='right')],
      dtype=object)
```

Reply:
For a big array, `pd.unique` is usually faster because it is unsorted. IIUC the type of `unique_values` doesn't matter, it gets stuck in to a IndeVariable `unique_coord` down below, so it will cast to numpy then anyway

Reply:
I agree that `pd.unique` is usually faster than `np.unique`. 

But now the comparison is `np.sort(pd.unique)` vs `np.unique`:

```python
a = np.random.randn(10000)

%timeit pd.unique(a)
468 µs ± 9.73 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit np.unique(a)
532 µs ± 5.01 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit np.sort(pd.unique(a))
1.03 ms ± 63.5 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Reply:
Right that's for all random entries. What about `a = pd.cut(np.random.randn(10000), 20)` which is more representative of binning  where we'd expect number of uniques << number of points.

Reply:
```python
a = pd.cut(np.random.randn(10000), 20)

%timeit pd.unique(a)
158 µs ± 3.68 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit np.unique(a)
70.1 ms ± 2.9 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit np.sort(pd.unique(a))
520 µs ± 14 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```
We have a winner then. :)